artix7/primitives/ff/ff.pb_type.xml

```diff
@@ -0,0 +1,239 @@
+          <T_setup    value="10e-12" port="BEL_FF-FDSE.CE" clock="C" />
+          <T_setup    value="10e-12" port="BEL_FF-FDSE.S" clock="C" />
+          <T_clock_to_Q max="10e-12" port="BEL_FF-FDSE.Q" clock="C" />
+        </pb_type>
+
+        <interconnect>
+          <direct name="D" input="BEL_FF-FDSE_or_FDRE.D" output="BEL_FF-FDSE.D" />
+          <direct name="CE" input="BEL_FF-FDSE_or_FDRE.CE" output="BEL_FF-FDSE.CE" />
+          <direct name="C" input="BEL_FF-FDSE_or_FDRE.C" output="BEL_FF-FDSE.C" />
+          <direct name="S" input="BEL_FF-FDSE_or_FDRE.SR" output="BEL_FF-FDSE.S" />
+          <direct name="Q" input="BEL_FF-FDSE.Q" output="BEL_FF-FDSE_or_FDRE.Q" />
+        </interconnect>
+      </mode>
+      <mode name="FDRE">
+        <pb_type name="BEL_FF-FDRE" num_pb="1" blif_model=".subckt FDRE">
+          <input  name="D" num_pins="1"/>
+          <input  name="CE" num_pins="1"/>
+          <clock  name="C" num_pins="1"/>
+          <input  name="R" num_pins="1"/>
+          <output name="Q" num_pins="1"/>
+          <T_setup    value="10e-12" port="BEL_FF-FDRE.D" clock="C" />
+          <T_setup    value="10e-12" port="BEL_FF-FDRE.CE" clock="C" />
+          <T_setup    value="10e-12" port="BEL_FF-FDRE.R" clock="C" />
+          <T_clock_to_Q max="10e-12" port="BEL_FF-FDRE.Q" clock="C" />
+        </pb_type>
+
+        <interconnect>
+          <direct name="D" input="BEL_FF-FDSE_or_FDRE.D" output="BEL_FF-FDRE.D" />
+          <direct name="CE" input="BEL_FF-FDSE_or_FDRE.CE" output="BEL_FF-FDRE.CE" />
+          <direct name="C" input="BEL_FF-FDSE_or_FDRE.C" output="BEL_FF-FDRE.C" />
+          <direct name="R" input="BEL_FF-FDSE_or_FDRE.SR" output="BEL_FF-FDRE.R" />
+          <direct name="Q" input="BEL_FF-FDRE.Q" output="BEL_FF-FDSE_or_FDRE.Q" />
+        </interconnect>
+
+        <metadata>
+          <meta name="fasm_features">ZRST</meta>
+        </metadata>
+      </mode>
+      <metadata>
+        <meta name="fasm_prefix">AFF BFF CFF DFF A5FF B5FF C5FF D5FF</meta>
+      </metadata>
+    </pb_type>
+
+    <interconnect>
+      <complete name="CE" input="BLK_BB-SLICE_FF.CE" output="BEL_FF-FDSE_or_FDRE.CE" />
+      <complete name="C" input="BLK_BB-SLICE_FF.CK" output="BEL_FF-FDSE_or_FDRE.C" />
+      <complete name="SR" input="BLK_BB-SLICE_FF.SR" output="BEL_FF-FDSE_or_FDRE.SR" />
+
+      <direct name="D" input="BLK_BB-SLICE_FF.D[3:0]" output="BEL_FF-FDSE_or_FDRE[3:0].D" />
+      <direct name="Q" input="BEL_FF-FDSE_or_FDRE[3:0].Q" output="BLK_BB-SLICE_FF.Q[3:0]" />
+
+      <direct name="D5" input="BLK_BB-SLICE_FF.D5[3:0]" output="BEL_FF-FDSE_or_FDRE[7:4].D" />
+      <direct name="Q5" input="BEL_FF-FDSE_or_FDRE[7:4].Q" output="BLK_BB-SLICE_FF.Q5[3:0]" />
+    </interconnect>
+    <metadata>
+      <meta name="fasm_features">FFSYNC</meta>
+    </metadata>
+  </mode>
+  <mode name="FDPE_or_FDCE">
+    <pb_type name="BEL_FF-FDPE_or_FDCE" num_pb="8">
```

Comment:
I think you actually want two sets of `<pb_type>` objects here, 4 * `nFF` and 4 * `n5FF`.